src/MassTransit/Context/LogContext.cs

```diff
@@ -0,0 +1,71 @@
+namespace MassTransit.Context
+{
+    using System;
+    using System.Diagnostics;
+    using Logging;
+    using Microsoft.Extensions.Logging;
+
+
+    public partial class LogContext
+    {
+        public static void ConfigureCurrentLogContext(ILoggerFactory loggerFactory = null, DiagnosticSource source = null)
+        {
+            Current = new BusLogContext(loggerFactory ?? NullLoggerFactory.Instance, source ?? Cached.Default.Value);
+        }
+
+        /// <summary>
+        /// Configure the current <see cref="LogContext"/> using the specified <paramref name="logger"/>, which will be
+        /// used for all log output.
+        /// </summary>
+        /// <param name="logger">An existing logger</param>
+        /// <param name="source">An optional custom <see cref="DiagnosticSource"/></param>
+        public static void ConfigureCurrentLogContext(Microsoft.Extensions.Logging.ILogger logger, DiagnosticSource source = null)
+        {
+            Current = new BusLogContext(new SingleLoggerFactory(logger), source ?? Cached.Default.Value);
+        }
+
+        public static ILogContext CreateLogContext(string categoryName)
```

Comment:
I really sorry to tell that, but I do not like static approach, could we create LogContext when creating a bus, and on every Consume/Send/Publish/etc create nested LogContext and inject it into, I'd love to say Container, but for example in MessageContext?